src/main/java/org/ngafid/FindLowEndingFuelEvents.java

```diff
@@ -0,0 +1,173 @@
+package org.ngafid;
+
+import org.ngafid.accounts.Fleet;
+import org.ngafid.common.TimeUtils;
+import org.ngafid.events.CustomEvent;
+import org.ngafid.events.EventDefinition;
+import org.ngafid.filters.Filter;
+import org.ngafid.flights.*;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.text.ParseException;
+import java.util.*;
+import java.util.logging.Logger;
+
+import static org.ngafid.events.CustomEvent.getLowEndFuelDefinition;
+import static org.ngafid.flights.CalculationParameters.*;
+
+public class FindLowEndingFuelEvents {
+    public static final Connection connection = Database.getConnection();
+    public static final Logger LOG = Logger.getLogger(FindLowEndingFuelEvents.class.getName());
+    private static Map<Integer, EventDefinition> eventDefs = new HashMap<>();
+    private static Map<Integer, Double> thresholds = new HashMap<>();
+
+
+    public static void findLowEndFuelEventsInUpload(Upload upload) {
+        try {
+            String whereClause = "upload_id = " + upload.getId() + " AND insert_completed = 1 AND NOT EXISTS (SELECT flight_id FROM events WHERE id = -4 OR id = -5 OR id = -6)";
```

Comment:
look at the where statement on line 598 in CalculateProximity.java and re-do this.

Reply:
also, make the -4, -5, -6 static variables (not hard coded).